File: docs/snippets/all/concepts/how-does-dalaran-work/log-to-grpc.cpp
#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    // Connect to the Dalaran gRPC server using the default address and
    // port: localhost:9876
    const auto rec = dalaran::RecordingStream("dalaran_example_log_to_grpc");
    rec.connect_grpc().exit_on_failure();

    // Log data as usual, thereby pushing it into the gRPC connection.
    while (true) {
        rec.log("log", dalaran::TextLog("Logging things…"));
    }
}

File: docs/snippets/all/concepts/how-does-dalaran-work/log-to-grpc.py
import dalaran as dl

dl.init("dalaran_example_log_to_grpc")

# Connect to the Dalaran gRPC server using the default address and url: dalaran+http://localhost:9876/proxy
dl.connect_grpc()

# Log data as usual, thereby pushing it into the gRPC connection.
while True:
    dl.log("/", dl.TextLog("Logging things…"))
